server/backend/testspg/router/test_router_user_role_assignment.py

```diff
@@ -428,103 +423,312 @@ def test_user_role_forest_client_number_not_exist_bad_request(
+
+    this test case uses business bceid user with FOM DEV FOM_SUBMITTER_00001018 privilege as example
+    test if business bceid user is delegated admin of FOM DEV for FOM_SUBMITTER role with forest client number 00001018,
+    able to remove FOM_SUBMITTER_00001018 access for business bceid user from same org,
+    will not be able to remove access with other forest client numbers like FOM_SUBMITTER_00001011
+    """
+    # create a user role assignment for a business bceid user within the same organization as the user COGNITO_USERNAME_BCEID
+    # with abstract role FOM_SUBMITTER and forest client number 00001018
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_ABSTRACT_BCEID,
+    )
+
+    # create a token for business bceid user COGNITO_USERNAME_BCEID with no app admin role
+    # we already grant this user the delegated admin privilege for role FOM_SUBMITTER_00001018
+    token = jwt_utils.create_jwt_token(
+        test_rsa_key, [], jwt_utils.COGNITO_USERNAME_BCEID
+    )
+    response = test_client_fixture.delete(
+        f"{endPoint}/{user_role_xref_id}",
+        headers=jwt_utils.headers(token),
+    )
+    # delete is successful
+    assert response.status_code == HTTPStatus.NO_CONTENT
+
+    # create a user role assignment for a business bceid user within the same organization as the user COGNITO_USERNAME_BCEID
+    # with abstract role FOM_SUBMITTER and forest client number 00001011
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        {**TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_ABSTRACT_BCEID, "forest_client_number": "00001011"},
+    )
+    response = test_client_fixture.delete(
+        f"{endPoint}/{user_role_xref_id}",
+        headers=jwt_utils.headers(token),
+    )
+    assert response.status_code == HTTPStatus.FORBIDDEN
+    assert response.json() is not None
+    data = response.json()
+    # business bceid user has no privilege to delete role with forest client number 00001011
+    assert data["detail"]["code"] == ERROR_PERMISSION_REQUIRED
+
+
+def test_deleter_user_role_assignment_bceid_cannot_delete_idir_access(
+    test_client_fixture: starlette.testclient.TestClient,
+    fom_dev_access_admin_token,
+    test_rsa_key,
+):
+    """
+    test business bceid user cannnot delete idir user access
+    """
+    # create a user role assignment for IDIR user
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_CONCRETE,
+    )
+
+    # create a token for business bceid user COGNITO_USERNAME_BCEID with FOM_DEV_ADMIN role
+    token = jwt_utils.create_jwt_token(
+        test_rsa_key, [FOM_DEV_ADMIN_ROLE], jwt_utils.COGNITO_USERNAME_BCEID
```

Comment:
Hi Catherine, curious that maybe we don't need `[FOM_DEV_ADMIN_ROLE]` to create the token?
Maybe there is no such case that the BCeID user can be assigned as App Admin (role in token) ?

Reply:
thanks Ian, originally I feel it's just for testing maybe doesn't matter, but I agree your concern. I updated, thanks!